fitparser/src/lib.rs

```diff
@@ -355,6 +357,97 @@ impl fmt::Display for ValueWithUnits {
     }
 }
 
+/// Developer fields are fields with properties that are not already defined
+/// by the SDK, but instead the definition is part of the fit file, for more
+/// flexiblity. Since their properties are dynamically defined and cannot be
+/// looked up statically we need to store these properties. This is what the
+/// DeveloperFieldDescription struct is for.
+///
+#[derive(Debug)]
+pub struct DeveloperFieldDescription {
+    developer_data_index: u8,
+    field_definition_number: u8,
+    fit_base_type_id: FitBaseType,
+    field_name: String,
+    scale: f64,
+    offset: f64,
+    units: String,
+}
+
+impl DeveloperFieldDescription {
+    /// Create DeveloperFieldDescription from FitDataFields
+    pub fn new(fields: &Vec<FitDataField>) -> Self {
```

Comment:
I'm not sure what would be best here, since we're processing "values" we could return some instances of `ErrorKind::ValueError` but then we'd need to change it from `new` to `try_from<Vec<FitDataField>>(...)` I think? Either way it would be good to move away from the panics.

Reply:
Using try_from as suggested now. All panics and expects are replaced with ErrorKind::ValueErrors